skbio/tree/_nj.py

```diff
@@ -216,15 +220,25 @@ def _lowest_index(dm):
     method (#228).
 
     """
-    lowest_value = np.inf
-    for i in range(dm.shape[0]):
-        for j in range(i):
-            curr_index = i, j
-            curr_value = dm[curr_index]
-            if curr_value < lowest_value:
-                lowest_value = curr_value
-                result = curr_index
-    return result
+    # convert to an array
+    dmarray = np.array(dm.to_data_frame())
```

Comment:
I don't think the cast to a `DataFrame` is necessary as the operations below should be able to leverage `dm.data` directly